### ISSUE TITLE
MessageBodyWriter not working properly

### ISSUE THREAD

**Issue by felipe-alves-moraes** [OpenLiberty/open-liberty#6041]
I have a project with the following `MessageBodyWriter`:
```
@Produces(MediaType.APPLICATION_JSON)
@Provider
public class EventMapper implements MessageBodyReader<Event>, MessageBodyWriter<Event> {

    @Context
    private Providers providers;
    private Jsonb jsonb;

    @PostConstruct
    private void init() {
        ContextResolver<Jsonb> resolver = providers.getContextResolver(Jsonb.class, MediaType.APPLICATION_JSON_TYPE);
        this.jsonb = resolver.getContext(Jsonb.class);
    }

    @Override
    public boolean isReadable(Class<?> aClass, Type type, Annotation[] annotations, MediaType mediaType) {
        return type == Event.class;
    }

    @Override
    public Event readFrom(Class<Event> aClass, Type type, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, String> multivaluedMap, InputStream inputStream) throws IOException, WebApplicationException {

        return jsonb.fromJson(inputStream, type);
    }

    @Override
    public boolean isWriteable(Class<?> aClass, Type type, Annotation[] annotations, MediaType mediaType) {
        return type == Event.class;
    }

    @Override
    public void writeTo(Event event, Class<?> aClass, Type type, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, Object> multivaluedMap, OutputStream outputStream) throws IOException, WebApplicationException {
        outputStream.write(jsonb.toJson(event).getBytes());
    }

}
```
For some reason, when I call the endpoint to retrieve an Event, I just receive an empty response `{}`, the same code works in other servers.

Here is the full code of my project: https://github.com/felipe-alves-moraes/events

**Comment by felipe-alves-moraes** [comment]
I was using OpenLiberty 18.0.0.3, and profile `javaee-8.0`. Changing to OpenLiberty 18.0.0.4, it's working the way I wanted, thanks @aguibert.

**Comment by aguibert** [comment]
Glad it works now! Closing out this issue as the problem is fixed in 18.0.0.4.
